libsolidity/Parser.cpp

```diff
@@ -691,9 +692,12 @@ ASTPointer<Expression> Parser::parsePrimaryExpression()
 		expression = nodeFactory.createNode<Literal>(token, getLiteralAndAdvance());
 		break;
 	case Token::NUMBER:
+		nextToken = m_scanner->peekNextToken();
 	case Token::STRING_LITERAL:
 		nodeFactory.markEndPosition();
-		expression = nodeFactory.createNode<Literal>(token, getLiteralAndAdvance());
+		expression = nodeFactory.createNode<Literal>(token, getLiteralAndAdvance(), nextToken);
```

Comment:
I think it would be cleaner to only pass nextToken to the constructor, if it is an EtherSubdenomination, i.e. pass a default value otherwise. Perhaps we could even already convert the enums here?